typescript/index.d.ts

```diff
@@ -64,26 +64,26 @@ export interface FormProps<FormValues = object>
   initialValuesEqual?: (a?: object, b?: object) => boolean;
 }
 
-export interface UseFieldConfig {
+export interface UseFieldConfig<FieldValue> {
   afterSubmit?: () => void;
   allowNull?: boolean;
   beforeSubmit?: () => void | boolean;
-  defaultValue?: any;
-  format?: (value: any, name: string) => any;
+  defaultValue?: FieldValue;
+  format?: (value: FieldValue, name: string) => any;
   formatOnBlur?: boolean;
-  initialValue?: any;
+  initialValue?: FieldValue;
   isEqual?: (a: any, b: any) => boolean;
   multiple?: boolean;
-  parse?: (value: any, name: string) => any;
+  parse?: (value: FieldValue, name: string) => FieldValue;
```

Comment:
This `value` parameter should still be `any`, I think, as it's what is returned from the input. i.e. say we had a field of type `number`, but we had it in an `<input type="text"/>`, the whole point of the `parse` function is to do that conversion. I don't think it's a common enough use case to provide an `UnparsedValue` generic type.

Reply:
Yea, that's true. Should have though about it a little bit more before I switched that. I'll switch it back to `any`.